### ISSUE TITLE
how to draw the heatmap？请问论文里的热力图是怎么做的啊？

### ISSUE THREAD

**Issue by hfutly66** [foundation-multimodal-models/CAL#2]

**Comment by Menoly-xin** [comment]
I apologize for the delayed response due to my ongoing graduation preparations and associated travels. For instructions on drawing the attention graph, please refer to the code below. You are supposed to set img tokens as -200 in your generation code.

`@torch.no_grad
def get_attention_score(outputs, input_ids, tokenizer):
    attn = sum(outputs["attentions"])
    attn = torch.log(attn)
    responce_start = torch.where(input_ids==13566)[1][0]
    img_s = torch.where(input_ids==-200)[1][0]
    response = attn[0, :, responce_start+2:-1]
    score = response[:, :, img_s: img_s+576]
    score = score.mean(dim=[0,1]).reshape(24, 24)
    return score`

**Comment by hfutly66** [comment]
Thank you very, very, very much. Have a good trip

**Comment by yuliang-liang** [comment]
Thanks.
